### ISSUE TITLE
Make localkube flag passing more robust

### ISSUE THREAD

**Issue by dlorenc** [kubernetes/minikube#512]
Right now minikube passes a set of flags to localkube via ssh at startup time. This works, but is problematic as we try to let minikube support multiple versions of localkube, and change the flags localkube takes over time. See #511 for one example of adding a flag.

Here are some ideas:

## Add a localkube command (or just parse the help text) to output what flags it takes.
Then minikube could first call that command, compare the list of valid flags with the list of flags the user wants to pass, and warn/error appropriately.

## Move flag passing to a config file
Then locallkube could only use the flags it knows about, and log a warning about ignored flags. This would be harder to get the warning back to the user automatically, but would avoid errors in the case of extra flags.

Any other ideas? Depending on what we do here, we might need to go back and re-release the last several localkube builds.

cc @aaron-prindle @r2d4

**Comment by dlorenc** [comment]
We could also consider reflection, to support dynamically configuring flags. Something like a config file to minikube with options like:

```yaml
kubernetesOptions:
  controllerManagerOptions:
    VolumeConfiguration.EnableHostPathProvisioning=true
```

would turn into setting:
`config.VolumeConfiguration.EnableHostPathProvisioning = true` in https://github.com/kubernetes/minikube/blob/master/pkg/localkube/controller-manager.go.

**Comment by dlorenc** [comment]
Rough reflection prototype here: https://play.golang.org/p/7IJfIz8AuP

It can set strings and ints on nested structs.

**Comment by aaron-prindle** [comment]
I believe this was fixed with #529. Closing.
